lace/emulator/nn_emulator.py

```diff
@@ -550,45 +708,68 @@ def train(self):
 
         loss_function_weights = self._set_weights()
         loss_function_weights = loss_function_weights.to(self.device)
+        
 
         log_kMpc_train = torch.log10(kMpc_train).to(self.device)
+        #log_kMpc_train = torch.log(kMpc_train).to(self.device)
 
-        self.log_kMpc = log_kMpc_train
+        self.log_kMpc = log_kMpc_train#[0]
+        
 
         self.nn = nn_architecture.MDNemulator_polyfit(
             nhidden=self.nhidden,
             ndeg=self.ndeg,
             max_neurons=self.max_neurons,
             ninput=len(self.emu_params),
         )
+        
+        if self.model_path is not None:
 
-        optimizer = optim.Adam(
-            self.nn.parameters(), lr=self.lr0, weight_decay=1e-4
-        )  #
+            pretrained_model = torch.load(
+                os.path.join(self.models_dir, self.model_path),
+                map_location="cpu",
+            )
+            self.nn.load_state_dict(pretrained_model["emulator"])
+            self.nn.to(self.device)
+            self.print("Loading pretrained initial state.")
+
+        optimizer = optim.AdamW(
```

Comment:
Given all these changes, I guess that optimizing the previous emulators (Cabayol23) would lead to different results, right? It is not a problem, just something that we should take into account

Reply:
Yes, sorry, this is something I wanted to make the distinction with the version in the paper and I forgot. Will add it.

Reply:
added! thanks!